lib/web_ui/lib/src/engine/text/line_breaker.dart

```diff
@@ -21,10 +21,41 @@ enum LineBreakType {
 }
 
 /// Acts as a tuple that encapsulates information about a line break.
+///
+/// It contains multiple indices that are helpful when it comes to measuring the
+/// width of a line of text.
+///
+/// [indexWithoutTrailingSpaces] <= [indexWithoutTrailingNewlines] <= [index]
 class LineBreakResult {
-  LineBreakResult(this.index, this.type);
-
+  LineBreakResult(
+    this.index,
+    this.indexWithoutTrailingNewlines,
+    this.indexWithoutTrailingSpaces,
+    this.type,
+  );
+
+  /// Creates a [LineBreakResult] where all indices are the same (i.e. there are
+  /// no trailing spaces or new lines).
+  LineBreakResult.sameIndex(this.index, this.type)
+      : indexWithoutTrailingNewlines = index,
+        indexWithoutTrailingSpaces = index;
+
+  /// The true index at which the line break should occur, including all spaces
+  /// and new lines.
   final int index;
+
+  /// The index of the line break excluding any trailing new lines.
+  final int indexWithoutTrailingNewlines;
+
+  /// The index of the line break excluding any trailing spaces.
+  final int indexWithoutTrailingSpaces;
```

Comment:
suggestion: would be nice to give example in doccomment here such as    lastword<sp><sp><cr><lf> specifying what the index value should be in in these examples.